db/schema_generator/nike_Ipsum_Generator.js

```diff
@@ -25,17 +25,17 @@ var sentenceGenerator = (minNumOfWords, maxNumOfWords, possiblyComma) => {
             : chosenWord + ' ';
     }
 
-    sentence = sentence[0].toUpperCase() + sentence.slice(1, sentence.length - 1) + '. ';
+    sentence = sentence[0].toUpperCase() + sentence.slice(1, sentence.length - 1) + '.';
         
     return sentence;
 }
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 var paragraphGenerator = (numOfSentences, minNumWords, maxNumWords, possiblyComma) => {
-    var paragraph = '';
+    var paragraph = [];
     for(var i = 0; i < numOfSentences; i++) {
-        paragraph+= sentenceGenerator(minNumWords, maxNumWords, possiblyComma);
+        paragraph.push(sentenceGenerator(minNumWords, maxNumWords, possiblyComma));
     }
-    paragraph = paragraph.substring(0, paragraph.length - 1);
+    paragraph = paragraph.join(' ');
     return paragraph;
 }
```

Comment:
This is nit-picky, but wouldn't the space/time complexity be less if you went back to the string method? Is there are reason for pushing to an array then joining? It seems like it could be adding an extra step that might not be necessary.

Reply:
yeah, im trying to remember why i changed it haha

mmm, but yeah i think you're right though. just add a space after each sentence, and then substring it.